system/Common.php

```diff
@@ -721,6 +721,22 @@ function is_really_writable(string $file): bool
     }
 }
 
+if (! function_exists('is_windows')) {
+    /**
+     * Detect if platform is running in Windows.
+     *
+     * @param bool|null $mock If given and a boolean then it will be the return value.
+     */
+    function is_windows(?bool $mock = null): bool
+    {
+        if (is_bool($mock)) {
+            return $mock;
+        }
```

Comment:
I don't think this will work for mocking. It would require every call to this function to pass the correct parameter. My proposal was more like this:
```suggestion
     * @param bool|null $mock Sets /clears the mock value to return for testing.
     */
    function is_windows(?bool $mock = null): bool
    {
        static $mocked;

        if (func_num_args()) {
            $mocked = $mock;
        }

        if (is_bool($mocked)) {
            return $mocked;
        }
```

Reply:
Yes, you are correct!